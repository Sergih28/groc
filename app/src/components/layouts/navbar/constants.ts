import type { NavBarElement } from './types'

export const NAVBAR_ELEMENTS: readonly NavBarElement[] = [
  {
    key: 'home',
    url: '/',
    isEnabled: true,
  },
  {
    key: 'pomodoro',
    url: '/pomodoro',
    isEnabled: false,
  },
  {
    key: 'notifications',
    url: '/notifications',
    isEnabled: false,
  },
  {
    key: 'settings',
    url: '/settings',
    isEnabled: false,
  },
  {
    key: 'stats',
    url: '/stats',
    isEnabled: false,
  },
  {
    key: 'challenges',
    url: '/challenges',
    isEnabled: false,
  },
  {
    key: 'export',
    url: '/export',
    isEnabled: false,
  },
  {
    key: 'help',
    url: '/help',
    isEnabled: false,
  },
] as const
