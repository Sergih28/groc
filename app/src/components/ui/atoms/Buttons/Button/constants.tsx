export const BUTTON_TYPES = {
  button: 'button',
  submit: 'submit',
  reset: 'reset',
} as const

export const BUTTON_STYLES = {
  general: 'general',
  delete: 'delete',
  submit: 'submit',
  counter: 'counter',
  options: 'options',
} as const
