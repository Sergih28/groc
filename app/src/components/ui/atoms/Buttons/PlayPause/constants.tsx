export const DEFAULT_TEXT = 'Play'
