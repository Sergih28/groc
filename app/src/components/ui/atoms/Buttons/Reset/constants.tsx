import { getLangFromWindowUrl, useTranslations } from '@i18n/utils'

const language = getLangFromWindowUrl()
const t = useTranslations(language)

export const RESET_BUTTON_TEXT = t('button.reset')
