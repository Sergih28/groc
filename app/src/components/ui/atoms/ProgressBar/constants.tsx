export const DEFAULT_PROGRESSBAR_PROPS = {
  backgroundColor: 'rgb(0, 0, 255)',
  fillColor: 'rgb(255, 255, 0)',
  baseValue: 0,
  targetValue: 100,
} as const
