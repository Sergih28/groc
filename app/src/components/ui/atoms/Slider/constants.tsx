export const DEFAULT_SLIDER_VALUES = {
  INITIAL: 0,
  MAX: 25,
  MIN: 0,
} as const
