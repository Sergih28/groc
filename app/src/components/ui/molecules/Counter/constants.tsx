export const BUTTON_TEXT = {
  START: 'Start',
  PAUSE: 'Pause',
  CONTINUE: 'Continue',
}
