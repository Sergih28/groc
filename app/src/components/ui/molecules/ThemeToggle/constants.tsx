export const THEMES = {
  dark: true,
  light: false,
} as const

export const DEFAULT_THEME = THEMES.light

export const THEMES_TEXT = {
  dark: 'dark',
  light: 'light',
} as const

export const THEME_TOGGLE_TEXT = {
  dark: 'ðŸŒž',
  light: 'ðŸŒ™',
}
