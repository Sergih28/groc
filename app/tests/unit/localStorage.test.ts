import * as sampleData from '@data/pomodoroSampleData.json'
import { DEFAULT_PHASE, DEFAULT_SETTINGS_VALUES } from '@store/constants'
import { localStorageItems } from '@utils/storage/keys'
import {
  createPomodoro,
  deletePomodoro,
  generatePomodoro,
  getLastPomodoro,
  getPomodoros,
  updatePomodoros,
} from '@utils/storage/pomodoro'
import type { PomodoroType } from '@utils/storage/types'

describe('localStorage tests', () => {
  const mockedRandomUUID = '1838f290-5dfc-478b-b21c-6d867ff1223a'
  beforeAll(() => {
    crypto.randomUUID = vi.fn(() => mockedRandomUUID)
  })

  afterAll(() => {
    vi.resetAllMocks()
  })

  afterEach(() => {
    localStorage.clear()
  })

  test('localStorage keys snapshot', () => {
    expect(localStorageItems).toMatchSnapshot()
    expect(crypto.randomUUID()).toBe(mockedRandomUUID)
  })

  describe('getPomodoros()', () => {
    test('given no pomodoro data in localStorage, should return an empty array', () => {
      const pomodoroData = getPomodoros()

      expect(pomodoroData).toEqual([])
    })

    test('given pomodoro list data in localStorage, should return an array of pomodoros objects', () => {
      const pomodoro: PomodoroType[] = [
        {
          id: crypto.randomUUID(),
          startTime: Date.now(),
          endTime: null,
          pausedTimeRanges: null,
          expectedDuration: 25,
          phase: 'pomodoro',
        },
      ]

      localStorage.setItem(localStorageItems.pomodoro, JSON.stringify(pomodoro))

      const savedPomodoros = getPomodoros()

      expect(savedPomodoros).toEqual([...pomodoro])
    })
  })

  describe('createPomodoro()', () => {
    test('creates a new pomodoro object with default values', () => {
      const newPomodoro = generatePomodoro()

      expect(newPomodoro).toBeDefined()
      expect(newPomodoro).toEqual({
        id: crypto.randomUUID(),
        startTime: newPomodoro.startTime,
        endTime: null,
        pausedTimeRanges: null,
        phase: DEFAULT_PHASE,
        expectedDuration: DEFAULT_SETTINGS_VALUES[`${DEFAULT_PHASE}Duration`],
      })
    })
  })

  describe('deletePomodoro()', () => {
    test('given a pomodoro id, it should delete that given pomodoro from the localStorage', () => {
      const pomodoro = generatePomodoro()
      createPomodoro(pomodoro)

      let savedPomodoros = getPomodoros()

      expect(savedPomodoros).toEqual([pomodoro])

      deletePomodoro(pomodoro.id)

      savedPomodoros = getPomodoros()

      expect(savedPomodoros).toEqual([])
    })
  })

  describe('updatePomodoros()', () => {
    test('given a new array of pomodoros, should update the localStorage to that given list', () => {
      const pomodoroList = sampleData as PomodoroType[]

      let savedPomodoros = getPomodoros()

      expect(savedPomodoros).toEqual([])

      updatePomodoros(pomodoroList)

      savedPomodoros = getPomodoros()

      expect(savedPomodoros).toMatchObject(pomodoroList)
    })
  })

  describe('getLastPomodoro()', () => {
    test('should return null if there is no pomodoro data', () => {
      const pomodoro = getLastPomodoro()

      expect(pomodoro).toBeNull()
    })
  })

  test('should return the last pomodoro if there is data', () => {
    const newPomodoro = generatePomodoro()
    createPomodoro(newPomodoro)

    const lastPomodoro = getLastPomodoro()

    expect(lastPomodoro).not.toBeNull()
    expect(lastPomodoro).toBeDefined()
  })
})
