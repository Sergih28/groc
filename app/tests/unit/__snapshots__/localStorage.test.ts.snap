// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`localStorage tests > localStorage keys snapshot 1`] = `
{
  "darkMode": "darkMode",
  "pomodoro": "pomodoro",
}
`;
